consul/src/main/scala/io/buoyant/consul/v1/BaseApi.scala

```diff
@@ -27,8 +27,11 @@ trait BaseApi extends Closable {
     RetryPolicy.backoff(backoffs) {
       // We will assume 5xx are retryable, everything else is not for now
       case (_, Return(rep)) => rep.status.code >= 500 && rep.status.code < 600
+      // Looks like it could be either of 2 depending on where FutureCancelledException occurred
+      case (_, Throw(e: Failure)) if e.isFlagged(Failure.Interrupted) => false
+      case (_, Throw(e: FutureCancelledException)) => false
```

Comment:
from a `Var.async` we return a closable:
```
      Closable.make { _ =>
        running = false
        pending.raise(new FutureCancelledException)
        Future.Unit
      }
```
Is there a more idiomatic way to cancel a future?